erpnext/accounts/doctype/pricing_rule/pricing_rule.py

```diff
@@ -217,7 +217,7 @@ def get_pricing_rule_for_item(args, price_list_rate=0, doc=None, for_validate=Fa
 		doc = frappe.get_doc(doc)
 
 	if (args.get('is_free_item') or
-		args.get("parenttype") == "Material Request"): return {}
+		args.get("parenttype") in ["Material Request", "Purchase Order"]): return {}
```

Comment:
this is not the best way to fix it since it wont fetch pricing rule for Purchase Order, and it needs to. Material Request does not give priority to fiscal matters, but Purchase Order does.
The issue is that the context of the parent form is being passed due to which the header gets copied. 
get_pricing_rule_for_item is also called via get_item_details when an item is set in the child row, but it doesnt break there becuase the context of the child row is passed perhaps.
Will have to find a solution on client side for this